client/src/components/LinkButton.jsx

```diff
@@ -1,56 +1,47 @@
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import propTypes from 'prop-types';
-
-import { useLink } from '../services';
+import Button from 'plaid-threads/Button';
+import { usePlaidLink } from 'react-plaid-link';
 
 LinkButton.propTypes = {
-  userId: propTypes.number,
-  itemId: propTypes.number,
-  altClasses: propTypes.string,
-  primary: propTypes.bool,
+  config: propTypes.object,
+  update: propTypes.bool,
 };
 
 LinkButton.defaultProps = {
-  userId: null,
-  itemId: null,
-  altClasses: null,
-  primary: false,
+  config: null,
+  update: false,
 };
 
-export default function LinkButton({
-  children,
-  userId,
-  itemId,
-  altClasses,
-  primary,
-}) {
-  const [linkInstance, setLinkInstance] = useState({});
-  const { linkHandlers, getLinkHandler } = useLink();
-
-  const isPrimary = primary ? 'button--is-primary' : '';
-  const classlist = altClasses !== null ? altClasses : '';
-
-  useEffect(() => {
-    getLinkHandler({ userId, itemId });
-  }, [userId, itemId, getLinkHandler]);
-
-  useEffect(() => {
-    if (itemId) {
-      setLinkInstance(linkHandlers.byItem[itemId] || {});
-    } else {
-      setLinkInstance(linkHandlers.byUser[userId] || {});
-    }
-  }, [linkHandlers, itemId, userId]);
+export default function LinkButton({ children, config, update }) {
+  const { open, ready } = usePlaidLink(config);
 
   return (
-    <button
-      className={`button ${isPrimary} ${classlist}`}
-      disabled={!linkInstance.isReady}
-      onClick={() => {
-        linkInstance.handler.open();
-      }}
-    >
-      {children}
-    </button>
+    <>
+      {!update ? (
+        <Button
+          centered
+          inline
+          small
+          disabled={!ready}
+          onClick={() => {
+            open();
+          }}
+        >
+          {children}
+        </Button>
+      ) : (
+        // case where link is launched in update mode from dropdown menu in the
+        // item card after item is set to "bad state"
+        <div
+          disabled={!ready}
+          onClick={() => {
```

Comment:
`div`s cant have an `onClick`. We can use a threads Button, Touchable, or InlineLink or any other clickable element instead

Reply:
oops, yep, i missed this one. I will make it into a button.